sql-plugin/src/main/scala/com/nvidia/spark/rapids/literals.scala

```diff
@@ -98,12 +247,18 @@ object GpuScalar {
       }
     case dec: BigDecimal =>
       Scalar.fromDecimal(-dec.scale, dec.bigDecimal.unscaledValue().longValueExact())
+    case _: ArrayData =>
+      // Not sure why there are two independent `from`s. A little confused that why this one
+      // is needed since Spark has already inferred the DataType for the literal value.
+      throw new IllegalStateException("Please calls 'from(Any, DataType)' instead to" +
+        " create a Scalar of array type.")
     case _ =>
       throw new IllegalStateException(s"${v.getClass} '$v' is not supported as a scalar yet")
   }
 
   def from(v: Any, t: DataType): Scalar = v match {
-    case _ if v == null => Scalar.fromNull(GpuColumnVector.getNonNestedRapidsType(t))
+    case _ if v == null => Scalar.fromNull(GpuColumnVector.toRapidsType(t))
```

Comment:
So I got a question.
How to define a null LIST Scalar ?  Besides setting the validity flag to `false`, we still need to create an empty column with the correct children for it, instead of the default column ?

Reply:
Yes, this is why the c++ code does not let you make an empty nested column with just the data_type.

https://github.com/rapidsai/cudf/blob/2ead87c7d0245e5fef08c6fde44c8f7d978874ac/cpp/src/column/column_factories.cpp#L67-L68

But instead wrote `cudf::empty_like` so you could make an empty column that looks like an existing one.

https://github.com/rapidsai/cudf/blob/2ead87c7d0245e5fef08c6fde44c8f7d978874ac/cpp/include/cudf/copying.hpp#L177-L183

And also why we check that when we get a column that all of the children columns exist and are the proper types.

https://github.com/NVIDIA/spark-rapids/blob/c6931355fc32736ba1e7383dbcd0dfee4ca760c1/sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java#L616-L622

Otherwise we risk producing an invalid memory layout at some point. The fact that you could make a list scalar with no column view feels really bad to me.

Reply:
Filed a PR https://github.com/rapidsai/cudf/pull/8173 to support this.